### DISCUSSION TITLE
Docker Desktop 4.28.0+ incompatible with PrairieLearn local development on Windows

### DISCUSSION THREAD

**Discussion by nwalters512** [PrairieLearn/PrairieLearn#9738] [Announcements]
Docker Desktop 4.28.0 broke functionality required for the local development of PrairieLearn external graders and workspaces. Specifically, it does not correctly handle mounting volumes when a container is created via the Docker socket that is mounted into another container.

- https://github.com/docker/for-win/issues/13977
- https://github.com/docker/for-win/issues/13980

If you are running Windows and experience issues while locally running external graders or workspaces, downgrading to [Docker Desktop 4.27.2](https://docs.docker.com/desktop/release-notes/#4272) should help.

**Comment by jonatanschroeder** [comment]
Note that running the installer if you already have a newer version will not allow you to downgrade by default. You can downgrade by opening a terminal and typing:
```
.\DockerDesktopInstaller.exe install --disable-version-check
```
Credit: https://stackoverflow.com/a/77224786

**Comment by jonatanschroeder** [comment]
This is resolved in 4.30.0: https://docs.docker.com/desktop/release-notes/#for-windows-1.

> Fixed a regression in Docker Desktop 4.28 that caused host files bind-mounted into containers to not show up properly inside the container, when using Docker-in-Docker (via mounts of /var/run/docker.sock) on WSL.

Tested locally and it indeed works.
